src/newshop.cpp

```diff
@@ -113,16 +113,16 @@ bool is_ok_char(struct char_data * keeper, struct char_data * ch, vnum_t shop_nr
   if ((shop_table[shop_nr].races.IsSet(RACE_HUMAN) && GET_RACE(ch) == RACE_HUMAN) ||
```

Comment:
Do we need DRAKE_HUMAN and GHOUL_HUMAN in here?

Reply:
> if ((shop_table[shop_nr].races.IsSet(RACE_HUMAN) && GET_RACE(ch) == RACE_HUMAN) || GET_RACE(ch) == RACE_DRAKE_HUMAN) ||

Little sketchy here, since it'll spit out any error with this additional syntax

Reply:
The logic here is a mess, I need to rewrite this whole check some day. In the meantime, you're getting an error because your conditions are grouped wrong. Try:
```c++
if ((shop_table[shop_nr].races.IsSet(RACE_HUMAN) && (GET_RACE(ch) == RACE_HUMAN || GET_RACE(ch) == RACE_DRAKE_HUMAN || GET_RACE(ch) == RACE_GHOUL_HUMAN)) ||
```

Reply:
Ah, I see. Should be fixed, I also dropped ghouls from the table in favour of the YES_GHOUL shop_flag